packages/cli-hermes/src/profileHermes/downloadProfile.ts

```diff
@@ -0,0 +1,128 @@
+import {Config} from '@react-native-community/cli-types';
+import {execSync} from 'child_process';
+import {logger, CLIError} from '@react-native-community/cli-tools';
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import transformer from 'hermes-profile-transformer';
+import {findSourcemap, generateSourcemap} from './sourcemapUtils';
+import {
+  getAndroidProject,
+  getPackageName,
+} from '@react-native-community/cli-platform-android';
+/**
+ * Get the last modified hermes profile
+ * @param packageName
+ */
+function getLatestFile(packageName: string): string {
+  try {
+    const file = execSync(`adb shell run-as ${packageName} ls cache/ -tp | grep -v /$ | egrep '\.cpuprofile' | head -1
+        `);
+    return file.toString().trim();
+  } catch (e) {
+    throw new Error(e);
+  }
+}
+
+function execSyncWithLog(command: string) {
+  logger.debug(`${command}`);
+  return execSync(command);
+}
+
+/**
+ * Pull and convert a Hermes tracing profile to Chrome tracing profile
+ * @param ctx
+ * @param dstPath
+ * @param fileName
+ * @param sourceMapPath
+ * @param raw
+ * @param generateSourceMap
+ */
+export async function downloadProfile(
+  ctx: Config,
+  dstPath: string,
+  fileName?: string,
+  sourcemapPath?: string,
+  raw?: boolean,
+  shouldGenerateSourcemap?: boolean,
+  port?: string,
+) {
+  try {
+    const androidProject = getAndroidProject(ctx);
```

Comment:
If we can't find Android project or get package name - we can provide descriptive error message.
For example, when I run command outside project I get error "error Failed to download the Hermes Sampling Profile" which doesn't say what I did wrong. Especially that I knew I had cpuprofiles on a device.

Reply:
Hi @likern, thanks a lot for reviewing my PR! I looked at the code again, and think that the functions `getAndroidProject` and `getPackageName` already handled the error (specific details in my comments above). So can you check again whether any other issue happened when you run the command outside the project?

Reply:
@jessieAnhNguyen I'm not sure is it my fault or not.
I have build package using `yarn build` and later run cli directly through `node /home/me/react-native-cli/packages/cli/build/bin.js`.

If I run inside project I see this:
```bash
[me@localhost Feeve]$ pwd 
/home/me/Breeffy/Feeve
[me@localhost Feeve]$ node /home/me/react-native-cli/packages/cli/build/bin.js profile-hermes
info Downloading a Hermes Sampling Profiler from your Android device...
info No filename is provided, pulling latest file
error Failed to download the Hermes Sampling Profile. Run CLI with --verbose flag for more details.
```

while if I run this command outside (in `$HOME` directory) this erro is shown:
```bash
[me@localhost ~]$ pwd
/home/me
[me@localhost ~]$ node ~/react-native-cli/packages/cli/build/bin.js profile-hermes
error Unrecognized command "profile-hermes".
info Run "react-native --help" to see a list of all available commands.
```

Not sure is it indended behaviour or not.

Reply:
If there is a specific way to build and run cli - please let me know, I will try to test again.

Reply:
Hey, can you try and test using the CLI's [testing plan](https://github.com/react-native-community/cli/blob/master/CONTRIBUTING.md#testing-your-changes). Please note that this suggests a way to link the packages, so that the React Native app that you're using to test on is linked to the CLI code. In the command `yarn link`, add this into the list of linked packages "@react-native-community/cli-hermes"

After that, you can simply run `react-native profile-hermes`. Please let me know if there's any issue!

Reply:
Hey @likern, I found the bug and fixed it. Essentially we were throwing our error in the functions but printing a custom message in the final catch block. It's fixed now, thank you so much for pointing this out!